lib/matplotlib/tests/test_axes.py

```diff
@@ -3826,9 +3826,14 @@ def test_stem(use_line_collection):
     fig, ax = plt.subplots()
     # Label is a single space to force a legend to be drawn, but to avoid any
     # text being drawn
-    ax.stem(x, np.cos(x),
-            linefmt='C2-.', markerfmt='k+', basefmt='C1-.', label=' ',
-            use_line_collection=use_line_collection)
+    if use_line_collection:
+        ax.stem(x, np.cos(x),
+                linefmt='C2-.', markerfmt='k+', basefmt='C1-.', label=' ')
+    else:
+        with pytest.warns(match='deprecated'):
+            ax.stem(x, np.cos(x),
+                    linefmt='C2-.', markerfmt='k+', basefmt='C1-.', label=' ',
+                    use_line_collection=False)
```

Comment:
This could be (with a suitable import):
```suggestion
    with (nullcontext() if use_line_collection else
          pytest.warns(match='deprecated')):
        ax.stem(x, np.cos(x),
                linefmt='C2-.', markerfmt='k+', basefmt='C1-.', label=' ',
                use_line_collection=use_line_collection)
```

Reply:
Ah, I missed it was not passed to avoid the deprecation.